Crash with "keys pull"

## Version of Singularity:

Head of master, 3.0.3-493.g07404c9c

### Expected behavior

Key should be pulled from library, or detect that key is already present

### Actual behavior

Crash, runtime error, SIGSEGV.

```
> singularity keys pull 1CEC42121A994A3FBFAAF6CC51BE5020C508C7E9
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1626797]

goroutine 1 [running]:
github.com/sylabs/singularity/cmd/internal/cli.doKeysPullCmd(0x7ffeefbff331, 0x28, 0x17beb2c, 0x16, 0x0, 0x0)
    /Users/mikegray/Documents/src/github.com/sylabs/singularity/cmd/internal/cli/keys_pull.go:66 +0x167
github.com/sylabs/singularity/cmd/internal/cli.glob..func8(0x1dbe120, 0xc000162bd0, 0x1, 0x1)
    /Users/mikegray/Documents/src/github.com/sylabs/singularity/cmd/internal/cli/keys_pull.go:31 +0x61
github.com/sylabs/singularity/vendor/github.com/spf13/cobra.(*Command).execute(0x1dbe120, 0xc0000c00f0, 0x1, 0x1, 0x1dbe120, 0xc0000c00f0)
    /Users/mikegray/Documents/src/github.com/sylabs/singularity/vendor/github.com/spf13/cobra/command.go:766 +0x2cc
github.com/sylabs/singularity/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x1dbc700, 0x0, 0xd7, 0x16a7001)
    /Users/mikegray/Documents/src/github.com/sylabs/singularity/vendor/github.com/spf13/cobra/command.go:852 +0x2fd
github.com/sylabs/singularity/vendor/github.com/spf13/cobra.(*Command).Execute(0x1dbc700, 0x4, 0x17da8a3)
    /Users/mikegray/Documents/src/github.com/sylabs/singularity/vendor/github.com/spf13/cobra/command.go:800 +0x2b
github.com/sylabs/singularity/cmd/internal/cli.ExecuteSingularity()
    /Users/mikegray/Documents/src/github.com/sylabs/singularity/cmd/internal/cli/singularity.go:118 +0x110
main.main()
    /Users/mikegray/Documents/src/github.com/sylabs/singularity/cmd/singularity/cli.go:22 +0x20
```

### Steps to reproduce behavior

`singularity keys pull 1CEC42121A994A3FBFAAF6CC51BE5020C508C7E9`

I do currently already have that key locally as I had verified the `library://alpine:latest` image - which is signed with that key.

```sh
$ singularity keys list
Public key listing (/Users/mikegray/.singularity/sypgp/pgp-public):

0) U: Mike Gray (staging keys) <mike@sylabs.io>
   C: 2018-10-06 12:14:26 -0400 EDT
   F: 8785520C79F09CB83C1A069A37FCD8D3B1E8121E
   L: 4096
   --------
1) U: Mike Gray (dd) <mike@mikegray.org>
   C: 2018-10-17 11:51:15 -0400 EDT
   F: 734C7B2922166DF3C94D5F7F6896225725AFF440
   L: 4096
   --------
2) U: Sylabs Admin () <support@sylabs.io>
   C: 2018-10-08 13:48:34 -0400 EDT
   F: 1CEC42121A994A3FBFAAF6CC51BE5020C508C7E9
   L: 4096
   --------
3) U: Mike Gray (mike-dev) <mike@mikegray.org>
   C: 2018-11-02 14:16:01 -0400 EDT
   F: 0651936B7564EA6EB0C6D36464DF7C8866A78792
   L: 4096
   --------
```

--- a/cmd/internal/cli/keys_pull.go
+++ b/cmd/internal/cli/keys_pull.go
@@ -62,12 +62,14 @@ func doKeysPullCmd(fingerprint string, url string) error {
 	defer fp.Close()
 
 	for _, e := range el {
+		storeKey := true
 		for _, estore := range elstore {
 			if e.PrimaryKey.KeyId == estore.PrimaryKey.KeyId {
-				e = nil // Entity is already in key store
+				storeKey = false // Entity is already in key store
+				break
 			}
 		}
-		if e != nil {
+		if storeKey {
 			if err = e.Serialize(fp); err != nil {
 				return err
 			}

